rest/admin_api.go

```diff
@@ -522,10 +522,25 @@ func (h *handler) handlePutDbConfig() (err error) {
 		}
 	}
 
-	bucket := h.db.Bucket.GetName()
+	// if the database is a corrupted database config then we will have no bucket on the db to use (ah teh db context
+	// was previously closed), so nil check needed here to protect against panic
+	// is needed
+	var bucket string
+	if h.db.Bucket != nil {
+		bucket = h.db.Bucket.GetName()
+	}
 	if dbConfig.Bucket != nil {
 		bucket = *dbConfig.Bucket
 	}
+	if bucket == "" {
+		// check if the db config is in corrupt state
+		if h.server.isInCorruptState(h.db.Name) {
+			// database is in corrupt state must provide bucket name to fix
+			return base.HTTPErrorf(http.StatusInternalServerError, "database is in corrupt state must provide bucket name in config update")
+		}
```

Comment:
I wonder if we can avoid storing the state of the corrupt databases, but we can do a read only when we do a put. Then we avoid having to worry about timing of when the bootstrap connection reads the config and we can avoid storing this as a global.